python_modules/dagster/dagster/_core/definitions/freshness_checks/time_partition.py

```diff
@@ -84,13 +118,169 @@ def build_time_partition_freshness_checks(
+            deadline = get_latest_completed_cron_tick(
+                deadline_cron, current_time_in_freshness_tz, timezone
+            )
+            deadline_in_partitions_def_tz = pendulum.from_timestamp(
+                deadline.timestamp(), tz=partitions_def.timezone
+            )
+            last_completed_time_window = check.not_none(
+                partitions_def.get_prev_partition_window(deadline_in_partitions_def_tz)
+            )
+            expected_partition_key = partitions_def.get_partition_key_range_for_time_window(
+                last_completed_time_window
+            ).start
+
+            last_update_time_lower_bound = check.not_none(
+                last_completed_time_window,
+                "Expected a valid partitioned asset with a completed time window in order to determine valid freshness window.",
+            ).end
+
+            latest_record = retrieve_latest_record(
+                instance=context.instance, asset_key=asset_key, partition_key=expected_partition_key
+            )
+            update_timestamp = get_last_updated_timestamp(latest_record, context)
+            passed = (
+                update_timestamp is not None
+                and update_timestamp >= last_update_time_lower_bound.timestamp()
+            )
+
+            metadata: Dict[str, MetadataValue] = {
+                FRESHNESS_PARAMS_METADATA_KEY: JsonMetadataValue(params_metadata),
+                LATEST_CRON_TICK_METADATA_KEY: TimestampMetadataValue(deadline.timestamp()),
+                LOWER_BOUND_TIMESTAMP_METADATA_KEY: TimestampMetadataValue(
+                    last_update_time_lower_bound.timestamp()
+                ),
+            }
+
+            # We use the latest record across all partitions to give the user additional information
+            # about the asset's state. Allows us to provide the last update time of the asset even
+            # in the failed case, and also to distinguish between the case where the asset hasn't
+            # been observed in a long time, and the case where the asset was actually observed
+            # recently, just with an old partition.
+            latest_record_any_partition = retrieve_latest_record(
+                instance=context.instance, asset_key=asset_key, partition_key=None
+            )
+            is_out_of_date_observation = (
+                is_observation_out_of_date(
+                    latest_record_any_partition, last_update_time_lower_bound
+                )
+                if latest_record_any_partition is not None
+                else False
+            )
+
+            if not passed and latest_record_any_partition is not None:
+                # If this asset has been updated at all before, provide the time at which that
+                # happened as additional metadata.
+                metadata[LAST_UPDATED_TIMESTAMP_METADATA_KEY] = TimestampMetadataValue(
```

Comment:
I think that including this metadata might be confusing and distracting in the partitioned case. Easy for users to get the idea that it affects the calculation, when it actually does not, right?

Reply:
Technically it currently does affect the calculation. I'm fine with changing it to not doing so, as I agree the value add is pretty low compared to the confusion it incurs.